File: python/shell.py
def shellSort(nums):
    n = len(nums)
    gap = int(n / 2)
    while gap > 0:
        for i in range(gap, n):
            j, num = i-gap, nums[i]
            while j >= 0 and nums[j] > num:
                nums[j+gap] = nums[j]
                j -= gap
            nums[j+gap] = num
        gap = int(gap / 2)
        
nums = [5, 4, 3, 6, 2, 1, 7]
shellSort(nums)
print(nums)

File: c++/shell/shell.cpp
#include <iostream>
using namespace std;


void shellSort(int nums[], int size){
    int gap = size / 2;
    for (; gap>0; gap/=2){
        for(int i=gap; i<size; i++){
            int j = i - gap;
            int num = nums[i];
            for (; j>=0 && nums[j]>num; j-=gap){
                nums[j+gap] = nums[j];
            }
            nums[j+gap] = num;
        }
    }
}


int main(){
    int nums[] = {4, 3, 6, 5, 1, 7, 2, 8, 9};
    int size = 9;
    shellSort(nums, size);
    for (int i=0; i<size; i++){
        cout<<nums[i]<<" ";
    }
    cout<<endl;
    system("pause");
    return 0;
}